src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -61,7 +63,7 @@ export default class DefaultMessagingSession implements MessagingSession {
   }
 
   start(): void {
-    if (this.isClosed()) {
+    if (this.isClosed() && !this.isConnecting) {
```

Comment:
Could you please explain why do we need the `isConnecting` here, is `start` getting called multiple times unintendedly?